### PR TITLE
Global Popup component

### PR DESCRIPTION
**Pull request by benjclark**
- Take global-article/js/popup.js as a starting point and rebuilds it in ES6 classes
- Uses Global-Expander (which deals with opening/closing/focusing)
- Cuts away some of the functionality that was present in the global-article popup because it was covered already by Global-Expander or was not deemed necessary. The key bit that has been retained is the calculation of where to position the popup, either above or below the popup trigger element based on how much space is available in the viewport.
- Has a data attribute API or can be used directly, same approach that we have taken with other components like Global-Expander
- Popups can be closed a variety of ways:
---- click outside of popup (Global-Expander covers this)
---- click on close button
---- window resize
---- keypress (space or enter) on close button
---- keypress (escape) whilst popup or contents are focused
---- tabbing out of the popup (Global-Expander covers this)

This PR only deals with the js. Plan to do another PR for the styles before version bumped to release version  1.0.0.

### CODE REVIEW

File: toolkits/global/packages/global-popup/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+```html
+<span data-popup data-popup-target="popupContent1">Popup trigger</span>
+<div id="popupContent1">
+    <p>Some popup text</p>
+</div>			
+```
+
+Three data attribute are required:
```
Comment (jon-stevens) [inline toolkits/global/packages/global-popup/README.md:27]: Should this be two?

File: toolkits/global/packages/global-popup/js/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	const triggers = makeArray(document.querySelectorAll('[data-popup]'));
+	if (triggers.length === 0) {
+		return;
+	}
+	triggers.forEach(trigger => {
+		if (!trigger.hasAttribute('data-popup-target')) {
+			return;
+		}
+		const targetElementSelector = trigger.getAttribute('data-popup-target');
+		if (!targetElementSelector) {
```
Comment (jon-stevens) [inline toolkits/global/packages/global-popup/js/index.js:18]: Is it worth doing a check for 
!document.querySelector(targetElementSelector)
as well

File: toolkits/global/packages/global-popup/js/popup.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	constructor(trigger, id) {
+		this._trigger = trigger;
+		this._id = id;
+		this._content = document.querySelector(`#${this._id}`);
+		this._className = 'c-popup';
+		this._isOpen = false;
+		this._expander = new Expander(this._trigger, this._content, {FOCUS_EVENT: true});
+		this._arrowClass = `${this._className}__arrow`;
+		this._closeClass = `${this._className}__close`;
+		this._closeButton = `<a href="javascript:;" class=${this._closeClass}>Close</a>`;
```
Comment (jon-stevens) [inline toolkits/global/packages/global-popup/js/popup.js:13]:
```suggestion
		this._closeButton = `<a href="javascript:;" class="${this._closeClass}">Close</a>`;
```

File: toolkits/global/packages/global-popup/js/popup.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+		this._arrow = `<div class=${this._arrowClass}></div>`;
```
Comment (jon-stevens) [inline toolkits/global/packages/global-popup/js/popup.js:14]:
```suggestion
		this._arrow = `<div class="${this._arrowClass}"></div>`;
```

File: toolkits/global/packages/global-popup/package.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@springernature/global-popup",
+  "version": "0.1.0",
+  "description": "Builds and styles a popup that can be opened and closed",
+  "license": "MIT",
+  "peerDependencies": {
+    "@springernature/global-context": "15.2.1"
```
Comment (ricky-lau) [inline toolkits/global/packages/global-popup/package.json:7]: Should switch to using `brand-content` as `global-content` is deprecated

**Pull request merged**